src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -5024,6 +5099,91 @@ private void PreVisitMetadataArg(CallNode node, TexlFunction func)
                 }
 
                 _txb.SetType(node, returnType);
+            }
+
+            // Method to previsit if callnode is determined as function with Type argument
+            private void PreVisitTypeArg(CallNode node, TexlFunction func)
+            {
+                AssertValid();
+                Contracts.AssertValue(node);
+                Contracts.AssertValue(func);
+                Contracts.Assert(func.HasTypeArgs);
+
+                var args = node.Args.Children.ToArray();
+                var argCount = args.Count();
+
+                Contracts.AssertValue(_txb.GetType(args[0]));
+
+                if (argCount < func.MinArity || argCount > func.MaxArity)
+                {
+                    ArityError(func.MinArity, func.MaxArity, node, argCount, _txb.ErrorContainer);
+                    _txb.SetInfo(node, new CallInfo(func, node));
+                    _txb.SetType(node, DType.Error);
+                    return;
+                }
+
+                Contracts.Assert(argCount > 1);
+                Contracts.AssertValue(args[1]);
+
+                if (args[1] is not FirstNameNode && args[1] is not TypeLiteralNode)

```

Comment:
Suggestion: move this after the ìf (args[1] is FirstNameNode typeName)` and `if (args[1] is TypeLiteralNode typeLiteral)` blocks - you can add it as a simple `else` block without any conditions.